src/core/datastores/persisters/key-value-persister.js

```diff
@@ -0,0 +1,67 @@
+const _cache = {};
+
+export class KeyValuePersister {
+  /** @type {Boolean} */
+  _cacheEnabled;
+  /** @type {Number} */
+  _ttl;
+
+  // TODO: implement TTL, make _cache a constructor argument?
+  constructor(cacheEnabled = true, ttl = Infinity) {
+    this._cacheEnabled = cacheEnabled;
+    this._ttl = ttl;
+  }
+
+  readEntities(key) {
+    if (this._cacheEnabled && _cache[key]) {
+      return Promise.resolve(_cache[key]);
+    }
+
+    return this._readFromPersistance(key)
+      .then((entities) => {
+        entities = entities || [];
+        if (this._cacheEnabled) {
+          _cache[key] = entities;
+        }
+        return entities;
+      });
+  }
+
+  persistEntities(key, entities) {
```

Comment:
We need to add some validation that will throw an `Error` if the `key` or `entities` arguments are `null` or `undefined`.